src/main/java/be/minelabs/data/MinelabsLangProvider.java

```diff
@@ -72,85 +75,37 @@ private Path getLangFilePath(String code) {
-        translationBuilder.add(Blocks.TIME_FREEZE_BLOCK, "");
-        translationBuilder.add(Blocks.TUBERACK, "");
-        translationBuilder.add(Blocks.UPQUARK_QUANTUMFIELD, "");
-        translationBuilder.add(Blocks.WEAK_BOSON, "");
-        translationBuilder.add(Blocks.WEAK_BOSON_QUANTUMFIELD, "");
-
-        // items (not BlockItems)
-        Items.ATOMS.forEach(atomItem -> translationBuilder.add(atomItem, ""));
-        translationBuilder.add(Items.MAGNET, "");
-        translationBuilder.add(Items.POCKET_HOLE, "");
-        translationBuilder.add(Items.LASERTOOL_IRON, "");
-        translationBuilder.add(Items.LASERTOOL_GOLD, "");
-        translationBuilder.add(Items.LASERTOOL_DIAMOND, "");
-
-        // entities (may include manual)
-        translationBuilder.add(Entities.ENTROPY_CREEPER, "");
-        translationBuilder.add(Entities.BALLOON, "");
-        translationBuilder.add("entity.minecraft.villager.sciencevillager", "");
-
+        // automatic Block/item/entity(type) key gen
+        addEmptyKeys(Registries.ITEM, (k, v) -> translationBuilder.add(v, EMPTY_TRANSLATION));
+        addEmptyKeys(Registries.BLOCK, (k, v) -> translationBuilder.add(v, EMPTY_TRANSLATION));
+        addEmptyKeys(Registries.ENTITY_TYPE, (k, v) -> translationBuilder.add(v, EMPTY_TRANSLATION));
+
+        // MANUALLY DOING GROUPS
+        translationBuilder.add(ItemGroups.ATOMS, EMPTY_TRANSLATION);
+        translationBuilder.add(ItemGroups.MINELABS, EMPTY_TRANSLATION);
+        translationBuilder.add(ItemGroups.QUANTUM_FIELDS, EMPTY_TRANSLATION);
+        translationBuilder.add(ItemGroups.CHEMICALS, EMPTY_TRANSLATION);
+        translationBuilder.add(ItemGroups.ELEMENTARY_PARTICLES, EMPTY_TRANSLATION);
+
+        translationBuilder.add("entity.minecraft.villager.sciencevillager", EMPTY_TRANSLATION);
+        // DO NOT USE
+        //addEmptyKeys(Registries.VILLAGER_PROFESSION, translationBuilder);
 
         // manual : text
-        translationBuilder.add("text.minelabs.valid", "");
-        translationBuilder.add("text.minelabs.invalid", "");
-        translationBuilder.add("text.minelabs.not_implemented", "");
-        translationBuilder.add("text.minelabs.multiple_molecules", "");
-        translationBuilder.add("text.minelabs.active", "");
-        translationBuilder.add("text.minelabs.inactive", "");
-        translationBuilder.add("text.minelabs.toggle_instruction", "");
-
+        translationBuilder.add("text.minelabs.valid", EMPTY_TRANSLATION);
+        translationBuilder.add("text.minelabs.invalid", EMPTY_TRANSLATION);
+        translationBuilder.add("text.minelabs.not_implemented", EMPTY_TRANSLATION);
+        translationBuilder.add("text.minelabs.multiple_molecules", EMPTY_TRANSLATION);
+        translationBuilder.add("text.minelabs.active", EMPTY_TRANSLATION);
+        translationBuilder.add("text.minelabs.inactive", EMPTY_TRANSLATION);
+        translationBuilder.add("text.minelabs.toggle_instruction", EMPTY_TRANSLATION);
```

Comment:
I don't see the value in adding these here. It's basically saying the code should generate missing values, so someone hopefully picks up on this and then copy pastes the values into the actual file. All the while you already knew they needed to be added so you could have just as easily put them in the right file in the first place :P.

Reply:
In case there is a new language added the default provider will generate all keys that we need.

Reply:
If that ever happens then we can copy paste an existing file and if the new lang file is opened in IntelliJ, it will tell us of missing values that do exist in the default language (en_us).

I just think it is simpler to add these fixed values in the json file rather than in code, also towards maintenance because not everyone knows about this generator so the two will likely diverge after a while.